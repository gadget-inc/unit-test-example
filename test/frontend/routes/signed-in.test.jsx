import { describe, expect, it, vi } from "vitest";
import renderer from "react-test-renderer";
import SignedIn from "../../../frontend/routes/signed-in";

// Gadget React hooks, such as useUser and useFindMany, can be mocked to return data
const mocks = vi.hoisted(() => {
  const defaultUser = {
    id: 0,
    email: "test@gmail.com",
    firstName: "Carl",
    lastName: "Weathers",
  };

  return {
    useUser: () => defaultUser,
    // vi.fn() allows us to set mock values in individual tests!
    useFindMany: vi.fn(),
  };
});

// mock the dependency using the hoisted mocks defined above
vi.mock("@gadgetinc/react", () => {
  return {
    useUser: mocks.useUser,
    useFindMany: mocks.useFindMany,
  };
});

// mock the api client in the file tree
vi.mock("../../../frontend/api", () => {
  return {
    api: {
      user: {},
      post: {},
    },
  };
});

// this is an example mocking out the Gadget API client and useUser hook
describe("snapshot test for frontend/routes/signed-in.jsx", () => {
  it("should render the default page for signed-in users", () => {
    // Provide mock data for the useFindMany
    mocks.useFindMany.mockReturnValue([
      {
        data: [
          {
            id: 1,
            title: "Test post",
          },
        ],
        fetching: false,
        error: null,
      },
    ]);

    const tree = renderer.create(<SignedIn />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render the loading message for signed-in users", () => {
    // Provide mock data for the useFindMany
    mocks.useFindMany.mockReturnValue([
      {
        data: null,
        fetching: true,
        error: null,
      },
    ]);

    const tree = renderer.create(<SignedIn />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
