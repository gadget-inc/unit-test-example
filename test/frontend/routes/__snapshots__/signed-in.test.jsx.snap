// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot test for frontend/routes/signed-in.jsx > should render the default page for signed-in users 1`] = `
[
  <div
    className="app-link"
  >
    <img
      alt="logo"
      className="app-logo"
      src="/frontend/assets/react-logo.svg"
    />
    <span>
      You are now signed into 
       
    </span>
  </div>,
  <div>
    <p
      className="description"
      style={
        {
          "width": "100%",
        }
      }
    >
      Start building your app's signed in area
    </p>
    <a
      href="/edit/files/frontend/routes/signed-in.jsx"
      rel="noreferrer"
      style={
        {
          "fontWeight": 500,
        }
      }
      target="_blank"
    >
      frontend/routes/signed-in.jsx
    </a>
  </div>,
  <div
    className="card-stack"
  >
    <div
      className="card"
    >
      <div
        className="card-content"
      >
        <img
          className="icon"
        />
        <div
          className="userData"
        >
          <p>
            id: 
            0
          </p>
          <p>
            name: 
            Carl
             
            Weathers
          </p>
          <p>
            email: 
            <a
              href="mailto:test@gmail.com"
            >
              test@gmail.com
            </a>
          </p>
          <p>
            created: 
          </p>
          <p>
            num posts: 
            1
          </p>
        </div>
      </div>
      <div
        className="sm-description"
      >
        This data is fetched from the user model
      </div>
    </div>
  </div>,
]
`;

exports[`snapshot test for frontend/routes/signed-in.jsx > should render the loading message for signed-in users 1`] = `
[
  <div
    className="app-link"
  >
    <img
      alt="logo"
      className="app-logo"
      src="/frontend/assets/react-logo.svg"
    />
    <span>
      You are now signed into 
       
    </span>
  </div>,
  <div>
    <p
      className="description"
      style={
        {
          "width": "100%",
        }
      }
    >
      Start building your app's signed in area
    </p>
    <a
      href="/edit/files/frontend/routes/signed-in.jsx"
      rel="noreferrer"
      style={
        {
          "fontWeight": 500,
        }
      }
      target="_blank"
    >
      frontend/routes/signed-in.jsx
    </a>
  </div>,
  <div
    className="card-stack"
  >
    <div
      className="card"
    >
      <div
        className="card-content"
      >
        <img
          className="icon"
        />
        <div
          className="userData"
        >
          <p>
            id: 
            0
          </p>
          <p>
            name: 
            Carl
             
            Weathers
          </p>
          <p>
            email: 
            <a
              href="mailto:test@gmail.com"
            >
              test@gmail.com
            </a>
          </p>
          <p>
            created: 
          </p>
          <p>
            Loading...
          </p>
        </div>
      </div>
      <div
        className="sm-description"
      >
        This data is fetched from the user model
      </div>
    </div>
  </div>,
]
`;
